service/refund.go

```
@@ -247,44 +247,76 @@ func (service *RefundService) UpdateGovPayBatchRefund(ctx context.Context, batch
 }
 
 // ProcessBatchRefund processes all refunds in the DB with a refund-pending status
-func (service *RefundService) ProcessBatchRefund(req *http.Request) (ResponseType, error) {
+func (service *RefundService) ProcessBatchRefund(req *http.Request) []string {
+	var errorList []string
 	payments, err := service.DAO.GetPaymentsWithRefundStatus()
 	if err != nil {
-		return Error, err
+		log.ErrorR(req, fmt.Errorf("error retrieving payments with refund-pending status: %w", err))
+		errorList = append(errorList, "error retrieving payments with refund-pending status")
+		return errorList
 	}
 	if len(payments) == 0 {
-		return NotFound, nil
+		log.DebugR(req, "no payments with refund-pending status found")
+		errorList = append(errorList, "no payments with refund-pending status found")
+		return errorList
 	}
 
 	for _, p := range payments {
 		if p.Data.PaymentMethod == "credit-card" {
-			return service.processGovPayBatchRefund(req, p)
+			err := service.processGovPayBatchRefund(req, p)
+			if err != "" {
+				errorList = append(errorList, err)
+			}
 		}
 	}
 
-	return Success, nil
+	return errorList
 }
 
-func (service *RefundService) processGovPayBatchRefund(req *http.Request, payment models.PaymentResourceDB) (ResponseType, error) {
+func (service *RefundService) processGovPayBatchRefund(req *http.Request, payment models.PaymentResourceDB) string {
```

This should return an error rather than a string